src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -3198,6 +3198,170 @@ internal static class UnsupportedWindows6_10_Ios_1_5_11
                     Join(GetFormattedString(MicrosoftNetCoreAnalyzersResources.PlatformCompatibilityFromVersionToVersion, "windows", "10.0", "12.0"), GetFormattedString(MicrosoftNetCoreAnalyzersResources.PlatformCompatibilityFromVersionToVersion, "ios", "9.0", "10.0"))));
         }
 
+        [Fact]
+        public async Task LocalFunctionWithoutAttributeWithinPropertyGetterNotWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+[SupportedOSPlatform(""Browser"")]
+public class Test
+{
+    [UnsupportedOSPlatform(""browser"")]
+    public static string CurrentProgram
+    {
+        get
+        {
+            return EnsureInitialized();
+
+            string EnsureInitialized() { return string.Empty; }
+        }
+    }
+}";
+            await VerifyAnalyzerAsyncCs(source, s_msBuildPlatforms);
+        }
+
+        [Fact]
+        public async Task UnsupportedContextReferencedApiWithinSameAssemblyAttributeNotWarn()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+[assembly:SupportedOSPlatform(""Browser"")] // Attribute from TFM target, can be ignored
+public class Test
+{
+    private string program;
+
+    [UnsupportedOSPlatform(""browser"")]
+    public string CurrentProgram
+    {
+        get
+        {
+            return program; // should not warn
+        }
+    }
+}";
+            await VerifyAnalyzerAsyncCs(source, s_msBuildPlatforms);
+        }
+
+        [Fact]
+        public async Task UnsupportedContextReferencedApiWithinSameAssemblyPlusImmediateAttributeWarns()
+        {
+            var source = @"
+using System.Runtime.Versioning;
+
+[assembly:SupportedOSPlatform(""Browser"")] // Attribute from TFM target, can be ignored
+public class Test
+{
+    [SupportedOSPlatform(""Browser"")] // Real platform dependency can be applied
+    private static string s_program;
+
+    [UnsupportedOSPlatform(""browser"")]
+    public static string CurrentProgram
+    {
+        get
+        {
+            return [|s_program|]; // should warn
+        }
+    }
+}";
+            await VerifyAnalyzerAsyncCs(source, s_msBuildPlatforms);
```

Comment:
We need to discuss this one in detail to determine if this logic makes sense. It illustrates the assembly-level attribute getting special treatment, which we'll want to be deliberate about solidifying in the rules.